infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/runtime/fsm/NegotiatingState.java

```diff
@@ -256,11 +264,20 @@ private void processNegotiationResponse(LogReplicationNegotiationResponse negoti
              * otherwise, start snapshot full sync.
              */
             if (logHead <= negotiationResponse.getLastLogProcessed() + 1) {
+                log.info("Resume LOG ENTRY sync. Address space has not been trimmed, deltas are guaranteed to be available. " +
+                        "logHead={}, lastLogProcessed={}", logHead, negotiationResponse.getLastLogProcessed());
                 fsm.input(new LogReplicationRuntimeEvent(LogReplicationRuntimeEvent.LogReplicationRuntimeEventType.NEGOTIATION_COMPLETE,
                         new LogReplicationEvent(LogReplicationEvent.LogReplicationEventType.REPLICATION_START,
                                 new LogReplicationEventMetadata(LogReplicationEventMetadata.getNIL_UUID(), negotiationResponse.getLastLogProcessed())
                         )));
             } else {
+                // TODO: it is OK for a first phase, but this might not be efficient/accurate, as the next (+1)
```

Comment:
@xiaoqin2012 can you please confirm this TODO note? To be sure keeping it around for future work.

Reply:
This is not written by me. Also the statement doesn't sound correct. If there are logs trimmed, we don't know if there are entries belong to the replicated tables or not. To be correct and safe, a full snapshot sync should be requested.

Reply:
It was written by me, as I was adding the logging and had never reviewed this code. I just wanted to have your input on this specific point. 
My point here is that getLastLogProcessed + 1 is not necessarily the next entry, right? so though getLastLogProcessed+1 falls in the trimmed space, maybe the next entry is getLastLogProcessed+2000 ( as this is a global address, which does not fall in the trimmed space). So, for better accuracy in the future (not now) we can leave this note around so we check the real next entry and avoid doing a full snapshot sync when not really required.

Reply:
Last processed is 10, now the log has been trimmed and the log head is 20. How do you know the entries between 11 and 19 contains the data to be replicated or not?

Reply:
We inspect txStream.next()and see the global address of the one coming next... and if it is 16, OK let's do snapshot sync. But if the next is 56, no need to.

Reply:
After trim, the log head is 20, the txStream's head at least will be 20 or bigger, right?
assume txStream has entries at 16, 18, 22. After trim, txStream's head will be 22, the next() call will give 22, right?

Reply:
mm, I see your point. But now I have a follow up question, the bitmaps keep track of the last trimmed address, so despite this scenario we could know that head is 22 and last trimmed address is 18. Again we don't need this now but I'm seeing if its an option.